Gems/AtomLyIntegration/CommonFeatures/Code/Source/CoreLights/PolygonLightDelegate.cpp

```diff
@@ -6,84 +6,92 @@
+                GetTransform().GetBasisZ());
         }
+    }
 
-        float PolygonLightDelegate::GetSurfaceArea() const
+    float PolygonLightDelegate::GetSurfaceArea() const
+    {
+        // Surprisingly simple to calculate area of arbitrary polygon assuming no self-intersection. See
+        // https://en.wikipedia.org/wiki/Shoelace_formula
+        float twiceArea = 0.0f;
+        AZStd::vector<Vector2> vertices = m_shapeBus->GetPolygonPrism()->m_vertexContainer.GetVertices();
+        for (size_t i = 0; i < vertices.size(); ++i)
         {
-            // Surprisingly simple to calculate area of arbitrary polygon assuming no self-intersection. See https://en.wikipedia.org/wiki/Shoelace_formula
-            float twiceArea = 0.0f;
-            AZStd::vector<Vector2> vertices = m_shapeBus->GetPolygonPrism()->m_vertexContainer.GetVertices();
-            for (size_t i = 0; i < vertices.size(); ++i)
-            {
-                size_t j = (i + 1) % vertices.size();
-                twiceArea += vertices.at(i).GetX() * vertices.at(j).GetY();
-                twiceArea -= vertices.at(i).GetY() * vertices.at(j).GetX();
-            }
-            float scale = GetTransform().GetUniformScale();
-            return GetAbs(twiceArea * 0.5f * scale * scale);
+            size_t j = (i + 1) % vertices.size();
+            twiceArea += vertices.at(i).GetX() * vertices.at(j).GetY();
+            twiceArea -= vertices.at(i).GetY() * vertices.at(j).GetX();
         }
+        float scale = GetTransform().GetUniformScale();
+        return GetAbs(twiceArea * 0.5f * scale * scale);
+    }
 
-        void PolygonLightDelegate::DrawDebugDisplay(const Transform& transform, const Color& color, AzFramework::DebugDisplayRequests& debugDisplay, bool isSelected) const
+    void PolygonLightDelegate::DrawDebugDisplay(
+        const Transform& transform, const Color& color, AzFramework::DebugDisplayRequests& debugDisplay, bool isSelected) const
+    {
+        if (isSelected)
         {
-            if (isSelected)
-            {
-                debugDisplay.SetColor(color);
+            debugDisplay.SetColor(color);
 
-                // Draw a Polygon for the attenuation radius
-                debugDisplay.DrawWireSphere(transform.GetTranslation(), GetConfig()->m_attenuationRadius);
-                debugDisplay.DrawArrow(transform.GetTranslation(), transform.GetTranslation() + transform.GetBasisZ());
-            }
+            // Draw a Polygon for the attenuation radius
+            debugDisplay.DrawWireSphere(transform.GetTranslation(), GetConfig()->m_attenuationRadius);
+            debugDisplay.DrawArrow(transform.GetTranslation(), transform.GetTranslation() + transform.GetBasisZ());
         }
+    }
 
-    } // namespace Render
-} // namespace AZ
+    Aabb PolygonLightDelegate::GetLocalVisualizationBounds() const
+    {
+        return Aabb::CreateCenterRadius(Vector3::CreateZero(), GetConfig()->m_attenuationRadius);
```

Comment:
I was surprised this needed to be a sphere, but unfortunately the polygon light doesn't draw a hemisphere for the standard one-sided case. I should fix that sometime. Looks like the same is true of quad lights too 😢 .

Reply:
Yeah I was just maintaining the existing behavior for these as close as possible. If it's relatively easy to add a hemisphere let me know and I can always try and fix that while I'm here, otherwise I can maybe make this a function that's referenced in both places so future changes will be easier to catch and not let the bounds and debug vis get out of sync.

Reply:
Hmmm.. there is actually a `DrawWireHemisphere` so it wouldn't be too hard to add the functionality for non-double-sided quad and poly lights. Up to you though, I get that's outside the scope of this change.

Reply:
I've left this as-is for now as I want to get this update complete in its current state but there are now tests to make updating this in future easier to know what's changed